src/app.rs

```diff
@@ -2539,12 +2539,21 @@ impl UiExtra for egui::Ui {
 }
 
 #[cfg(not(target_arch = "wasm32"))]
-pub fn start(data_sources: Vec<Box<dyn DeferredDataSource>>) {
+pub fn start(mut data_sources: Vec<Box<dyn DeferredDataSource>>) {
     env_logger::try_init().unwrap_or(()); // Log to stderr (if you run with `RUST_LOG=debug`).
 
+    let app_name = if data_sources.len() == 1 {
+        match data_sources[0].fetch_description().source_locator {
+            Some(source_locator) => source_locator,
+            _ => String::from("Legion Prof"),
+        }
+    } else {
+        String::from("Legion Prof")
+    };
```

Comment:
Perhaps this would be better as `"Legion Prof: <source locator>"` rather than just `"<source locator>"`?

Reply:
Yeah, some variation on this seems reasonable. `"Legion Prof: <source>` or `<source> - Legion Prof` both seem fine. The latter puts the source first, which may help if we're in a tab with limited title bar area.

Reply:
https://github.com/StanfordLegion/prof-viewer/pull/36/commits/c9504a6e3c2657bf479111cd2b4ae8dd7f2838d6

Reply:
What do you think should be the behavior when we have multiple data source? I see here you're using the first, which is fine. On the other hand, a large enough window could potentially show multiple sources, so we could concatenate them with spaces or something.

P.S. there's a typo in "Unknone".

Reply:
> What do you think should be the behavior when we have multiple data source?

I had thought about this, for a start I think we could display "Multiple Data Sources" which is informative without taking too much space. I guess I would be surprised if putting multiple sources in the title ended up being useful more often than not. Perhaps detailed information could be exposed somewhere else in the UI

Reply:
https://github.com/StanfordLegion/prof-viewer/pull/36/commits/7182c1a9cd9301ba2628adf9c23aa5df6f7a9352

Reply:
How about "<first source> and <N-1> other sources"? I think giving an impression of the total number of sources is useful.

Reply:
I suppose we could lay the groundwork in this PR with 
```rust
pub enum SourceLocator {
    Multiple(Vec<String>),
    Single(String),
    None,
}
```
Does that seem reasonable?

Reply:
Semantically, does this add much over a `Vec<String>`? The `Multiple` case presumably must always contain 2+ strings.

I believe you should be able to match against the number of elements in a `Vec`.

```rust
match locator[..] {
  [] => "No Source".to_string(),
  [x] => x,
  [x, ...] => format!("{} and {} other sources", x, locator.len()-1),
}
```

Reply:
I may have the syntax wrong on the `...`. It may be two dots.

https://pzol.github.io/getting_rusty/posts/20140417_destructuring_in_rust/#toc_2

Reply:
https://github.com/StanfordLegion/prof-viewer/pull/36/commits/12fb3ed00f705a7fbfd05b6417e29678ce5b6264

This commit coalesces any number of locators across any number of data sources

Reply:
Also https://github.com/StanfordLegion/prof-viewer/pull/36/commits/e418767647a1cba90543cfccd1df3b4e8a41cb7e to merge latest changes. @elliottslaughter do you expect different data sources could have identical locators? (i.e. should we dedup the list of locators on the backend side as well)

Reply:
I went ahead and added the uniquing here https://github.com/StanfordLegion/prof-viewer/pull/36/commits/aede088df7055909faa26ccd5ae5b31b2d9b1fb3

Reply:
I don't currently anticipate having duplicate data sources, but I don't think the uniquing hurts either. It's fine to leave it.